client/my-sites/email/titan-mail-quantity-selection/index.jsx

```diff
@@ -125,12 +129,21 @@ class TitanMailQuantitySelection extends React.Component {
 			} );
 	};
 
-	onQuantityChange = ( e ) => {
-		const parsedQuantity = parseInt( e.target.value, 10 );
+	onQuantityChange = ( event ) => {
+		const { target } = event;
+
+		// Handles the case where the user clicked on the 'Add another mailbox' button
+		if ( target instanceof HTMLSpanElement || target instanceof HTMLButtonElement ) {
+			this.setState( { quantity: this.state.quantity + 1 } );
+
+			return;
+		}
+
+		const parsedQuantity = parseInt( target.value, 10 );
 		const quantity = isNaN( parsedQuantity ) ? 1 : Math.max( 1, parsedQuantity );
 
-		if ( quantity.toString() !== e.target.value ) {
-			e.target.value = quantity;
+		if ( quantity.toString() !== target.value ) {
+			target.value = quantity;
 		}
```

Comment:
I'm not quite sure why we need this `if` clause. It would be great to add a comment explaining why we have to force the value of the input field.

Reply:
💯 RE the comment. I am also seeing some odd behaviour when entering values like `234.` and `123,12` - I believe this is related to my localised number representations, but in the first case the end result is `1`, and in the second the end result is `123` (which is at least reasonable).

But that's not what we're fixing in this PR...